api/server/routes_create_update.go

```diff
@@ -114,14 +114,13 @@ func (s *Server) ensureRoute(ctx context.Context, method string, wroute *models.
 }
 
 // ensureApp will only execute if it is on post or put. Patch is not allowed to create apps.
-func (s *Server) ensureApp(ctx context.Context, wroute *models.RouteWrapper, method string) error {
-	initApp := &models.App{Name: wroute.Route.AppID}
-	app, err := s.datastore.GetApp(ctx, initApp)
+func (s *Server) ensureApp(ctx context.Context, appName string, wroute *models.RouteWrapper, method string) error {
+	app, err := s.datastore.GetAppByName(ctx, appName)
```

Comment:
need to use id here too i think

Reply:
It's not possible with the current implementation. We use 1 handler to cover 3 different http methods and only one  of them doesn't require app to exist (PUT, POST call `ensureApp`, but PATCH doesn't do so). so, if there's no interest in splitting this handler into multiple ones there's no way to apply new middleware.

Reply:
the issue would be that we're ensuring potentially a different app than the one the front end got to begin with (to get to the handler) and even worse maybe updating it ?

Reply:
I don't think that there's possible case when you can ensure two different apps with this handler, the order of middlewares ensures that you still work with the same app as you requested to process.

Reply:
ensure app seems gone anyway now. however, UpdateApp is still getting the app by name in the transaction, which has this highlighted here.